--- a/include/string_theory/st_formatter.h
+++ b/include/string_theory/st_formatter.h
@@ -304,8 +304,15 @@ namespace ST
     ST_NODISCARD
     formatter_ref_t make_formatter_ref(type_T value)
     {
-        return [value](const ST::format_spec &format, ST::format_writer &output) {
+        return [value](const ST::format_spec &format, ST::format_writer &output) mutable {
             format_type(format, output, value);
         };
     }
+
+    inline void format_type(const ST::format_spec &format, ST::format_writer &output,
+                            std::string_view value)
+    {
+        format_type(format, output, std::string{value}.c_str());
+    }
+
 }
